src/Testing/TypeInferenceTestCase.php

```diff
@@ -126,12 +133,73 @@ public function assertFileAsserts(
 		}
 	}
 
+	public static function checkOnlyIfAnnotation(string $file): bool
+	{
+		$fileContents = file_get_contents($file);
+		if ($fileContents === false) {
+			self::fail(sprintf('Cannot read file %s', $file));
+		}
+
+		if (preg_match_all('{^(?:<\?php[^/\n]+)?//\s*onlyif\s+(.+)\s*$}Um', $fileContents, $ma) > 0) {
+			$onlyIfConditions = [];
+			for ($i = 0; $i < count($ma[0]); $i++) {
+				foreach ((array) preg_split('/\s*&&\s*/', $ma[1][$i]) as $onlyIfCondition) {
+					$onlyIfConditions[] = (string) $onlyIfCondition;
+				}
+			}
+
+			foreach ($onlyIfConditions as $onlyIfCondition) {
+				if (preg_match('/^(PHP_VERSION_ID|PHP_INT_SIZE) (>|>=|<|<=|==) ([0-9]+)$/', $onlyIfCondition, $m) > 0) {
+					$onlyIfConditionConstant = constant($m[1]);
+					$onlyIfConditionOperator = $m[2];
+					$onlyIfConditionValue = intval($m[3]);
```

Comment:
I think we should validate the type-range of the value. maybe you can re-use the typing PHPStan already knows for these 2 native constants

Reply:
Sure thing, I actually was always worried to miss a zero in 80000, so making sure it's at least 70000 and no more than 9999?

Reply:
yeah something like that.
- `PHP_INT_SIZE` can only be 4 or 8
- 99999 >= `PHP_VERSION_ID` >= 70200

Reply:
Added checks on the condition's value and unit tests